.gitlab/includes/hip_pipeline.yml

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2023 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include:
+  - local: '.gitlab/includes/common_pipeline.yml'
+  - local: '.gitlab/includes/common_spack_pipeline.yml'
+
+.variables_gcc12_hip5_config:
+  variables:
+    ARCH: linux-ubuntu22.04-zen3
+    BUILD_TYPE: Debug
+    COMPILER: gcc@12.2.0
+    CXXSTD: 20
+    GPU_TARGET: 'gfx90a'
+    SPACK_SPEC: "pika@main arch=$ARCH %${COMPILER} +rocm +stdexec amdgpu_target=${GPU_TARGET} \
+                 malloc=system cxxstd=$CXXSTD ^boost@1.79.0 ^hwloc@2.6.0 ^hip@5.5 ^llvm~gold \
+                 ^fmt@10.0.0 ^stdexec@git.nvhpc-23.09.rc4=main"
+
+gcc12_hip5_spack_image:
+  stage: spack_configs
+  timeout: 4 hours
+  needs: [base_spack_image]
+  extends:
+    - .container-builder
+    - .variables_gcc12_hip5_config
+  before_script:
+    - CONFIG_TAG=`echo $SPACK_SPEC-$SPACK_COMMIT | sha256sum - | head -c 16`
+    - compiler=${COMPILER/@/-}
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/pika-$compiler:$CONFIG_TAG
+    - echo -e "compiler=$compiler\nBASE_IMAGE=$PERSIST_IMAGE_NAME" >> compiler.env
+  variables:
+    BASE_IMAGE: $CSCS_REGISTRY_PATH/pika-spack-base:$SPACK_COMMIT
+    DOCKERFILE: .gitlab/docker/Dockerfile.spack_compiler
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE","ARCH","COMPILER","SPACK_SPEC"]'
+  artifacts:
+    reports:
+      dotenv: compiler.env
+
+gcc12_hip5_debug_build:
+  stage: build
+  timeout: 2 hours
```

Comment:
Is this necessary?

Reply:
Is this not the pika build stage (e.g. https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/479009878135925/5304355110917878/-/jobs/5870874330)? This step takes less than 10 minutes.